lib/mongo/client.rb

```diff
@@ -657,6 +648,14 @@ def update_options(new_options)
         @options = options.freeze
         validate_options!
         validate_authentication_options!
+
+        if @options.key?(:auto_encryption_options) && @options[:auto_encryption_options].nil?
```

Comment:
Can a comment be added explaining why the teardown happens when options are nil only?

Reply:
Sure!